File: e2e/selenide-junit5-selenium-grid/src/test/java/test/mobile/taskTable/TaskTableAccessibilityTest.java
package test.mobile.taskTable;

import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.TmsLink;
import test.base.MobileTest;
import test.spec.taskTable.ITaskTableAccessibilityTest;

import static data.AllureEpic.ACCESSIBILITY;

@Epic(ACCESSIBILITY)
@Feature("Task table")
@TmsLink("98")
class TaskTableAccessibilityTest extends MobileTest implements ITaskTableAccessibilityTest {}

File: demo-android/app/src/androidTest/java/com/example/demo/e2e/test/tasktable/TaskTableAccessibilityTest.kt
package com.example.demo.e2e.test.tasktable

import com.example.demo.e2e.context.TaskTestContext
import com.example.demo.e2e.data.AllureEpic.ACCESSIBILITY
import com.example.demo.e2e.test.base.AccessibilityTestBase
import io.qameta.allure.kotlin.Epic
import io.qameta.allure.kotlin.Feature
import io.qameta.allure.kotlin.TmsLink
import org.junit.Before
import org.junit.Test

@Epic(ACCESSIBILITY)
@Feature("Task table")
@TmsLink("98")
class TaskTableAccessibilityTest : AccessibilityTestBase() {
    @Before
    fun setup() {
        support().mock.api().getTasks(
            TaskTestContext().createTaskResponse().toTask(),
            TaskTestContext().createTaskResponse().toTask())
    }

    @Test
    fun shouldHaveNoAccessibilityViolationsOnTaskTableWhenTasksLoaded() {
        // Given
        steps.navigation.openMainPage()
        steps.navigation.refreshMainPage()

        // When
        steps.accessibility.analyze()

        // Then
        validate.accessibility.hasNoViolations()
    }
}
